### PR TITLE
Update ODL dependencies

### PR DESCRIPTION
**Pull request by biaol-odata**
### Issues

*This pull request updates the ODL dependencies to the newly release v7.5.0.*

### Description

*Update dependencies and make tests pass.*

### Checklist (Uncheck if it is not completed)

- [ ] *Test cases added*
- [ ] *Build and test with one-click build and test script passed*

### Additional work necessary

*If documentation update is needed, please add "Docs Needed" label to the issue and provide details about the required document change in the issue.*

### CODE REVIEW

File: test/E2ETest/Microsoft.Test.E2E.AspNet.OData/Build.AspNetCore/Microsoft.Test.E2E.AspNetCore.OData.csproj [hunk 1]
```diff
@@ -1721,17 +1721,17 @@
     <Reference Include="Microsoft.Net.Http.Headers, Version=2.0.1.0, Culture=neutral, PublicKeyToken=adb9793829ddae60, processorArchitecture=MSIL">
       <HintPath>..\..\..\..\sln\packages\Microsoft.Net.Http.Headers.2.0.1\lib\netstandard2.0\Microsoft.Net.Http.Headers.dll</HintPath>
     </Reference>
-    <Reference Include="Microsoft.OData.Client, Version=7.4.3.20321, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
-      <HintPath>..\..\..\..\sln\packages\Microsoft.OData.Client.7.4.3\lib\netstandard1.1\Microsoft.OData.Client.dll</HintPath>
+    <Reference Include="Microsoft.OData.Client, Version=7.5.0.20627, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
+      <HintPath>..\..\..\..\sln\packages\Microsoft.OData.Client.7.5.0\lib\net45\Microsoft.OData.Client.dll</HintPath>
```
Comment (xuzhg) [inline test/E2ETest/Microsoft.Test.E2E.AspNet.OData/Build.AspNetCore/Microsoft.Test.E2E.AspNetCore.OData.csproj]: It's changed to using "Net45"?

File: src/Microsoft.AspNet.OData.Shared/Query/Expressions/AggregationBinder.cs [hunk 2]
```diff
@@ -48,7 +48,9 @@ internal class AggregationBinder : ExpressionBinderBase
             {
                 case TransformationNodeKind.Aggregate:
                     var aggregateClause = this._transformation as AggregateTransformationNode;
+#pragma warning disable CS0618 // Type or member is obsolete
```
Comment (xuzhg) [inline src/Microsoft.AspNet.OData.Shared/Query/Expressions/AggregationBinder.cs]: don't use #pragma. 

I think you can change the codes as:

_aggregateExpressions = FixCustomMethodReturnTypes(aggregateClause.AggregateExpressions.OfType<AggregateExpression>());

Comment (kosinsky) [reply src/Microsoft.AspNet.OData.Shared/Query/Expressions/AggregationBinder.cs]: That's correct. In https://github.com/OData/WebApi/pull/906/files#diff-b18fa1821acbe9c329e386961320ad35 I'm changing fields to accept base class or using OfType

**Comment by biaol-odata** [comment]
Merged. Thanks for the CR!

**Pull request merged**